src/store/actions/open.js

```diff
@@ -76,15 +79,40 @@ export const openChannel = params => async (dispatch, getState, lh) => {
 
     requestBody.signed_tx = signed_tx;
 
+    // Timeout setup
+    const { chainId } = Lumino.getConfig().chainId;
+    const currentTimeout = TIMEOUT_MAP[chainId] || TIMEOUT_MAP[31];
+    // Cancellation setup
+    const CancelToken = Axios.CancelToken;
+    const source = CancelToken.source();
+
+    const timeoutId = setTimeout(() => {
+      Lumino.callbacks.trigger(
+        CALLBACKS.TIMED_OUT_OPEN_CHANNEL,
+        channel,
+        new Error("The operation took too much time")
+      );
+      source.cancel();
+    }, currentTimeout + 50);
```

Comment:
Why the addition of `50 `here?

Reply:
50 extra ms due to the possible time that we are wasting setting things up, but I can delete it.